solang-parser/src/lexer.rs

```diff
@@ -384,7 +384,7 @@ impl CodeLocation for LexicalError {
     }
 }
 
-static KEYWORDS: phf::Map<&'static str, Token> = phf_map! {
+pub static KEYWORDS: phf::Map<&'static str, Token> = phf_map! {
```

Comment:
Why is this public?

Reply:
I see you used this for finding keyword clashes, but I am inclined to think that this should not be a public symbol. Maybe `pub(crate)` would suffice.

Reply:
This is defined in the `solang-parser` crate, so making it `pub(crate)` is not going to work.

Reply:
You're parsing the IDL file too. Shouldn't this be part of `solang-parser?

Reply:
I don't agree with that. IDL is json, so there isn't any parsing involved - at least no custom code, the definition is entirely imported from the `anchor-syn` crate. `solang-parser` just contains a Solidity parser, that is totally unrelated to generating Solidity from IDL. How would this benefit Foundry for example?

I have created a new function `is_keyword()` in `solang-parser`, maybe this is a nicer interface.